mod/ArchConsoleManager.cs

```diff
@@ -187,23 +195,31 @@ public void AddText(string text, bool skipGameplayConsole = false, AudioType sou
             // If the consoles haven't been created yet, then adding to history is all we want to do for now.
             if (pauseConsoleText == null) return;
 
-            string consoleText = pauseConsoleText.text;
-            if (consoleText == "")
+            // bufferedText = pauseConsoleText.text;
+            if (bufferedText == "")
             {
-                consoleText = text;
+                bufferedText = text;
             }
             else
-            {                
-                consoleText += "\n" + text;
+            {
+                bufferedText += "\n" + text;
             }
             // Overflow fix
-            while (consoleText.Length > maxCharacters)
+            while (bufferedText.Length > maxCharacters)
             {
-                string str = consoleText.Split('\n')[0] + "\n";
-                consoleText = consoleText.Replace(str, "");
+                string str = bufferedText.Split('\n')[0] + "\n";
+                bufferedText = bufferedText.Replace(str, "");
                 overflowWarning.SetActive(true);
             }
-            pauseConsoleText.text = consoleText;
+            // Only bother updating the pause console text if the game is paused, hopefully reducing Layout calls
+            if (isPaused)
+            {
+                pauseConsoleText.text = bufferedText;
```

Comment:
Doesn't this set the pause console to show only messages we've received since the last pause?